packages/ag-charts-community/src/chart/legend/legendEvent.ts

```diff
@@ -0,0 +1,42 @@
+import type { AgChartLegendClickEvent, AgChartLegendDoubleClickEvent } from 'ag-charts-types';
+
+import { defineSetterGetter } from '../../util/object';
+
+export function makeLegendItemEvent<T extends 'click'>(
+    type: T,
+    itemId: string,
+    seriesId: string,
+    event: Event
+): AgChartLegendClickEvent;
+
+export function makeLegendItemEvent<T extends 'dblclick'>(
+    type: T,
+    itemId: string,
+    seriesId: string,
+    event: Event
+): AgChartLegendDoubleClickEvent;
+
+export function makeLegendItemEvent(
```

Comment:
Potentially we could change the legend click event to include `type: 'click' | 'dblclick'`, and we could just instantiate a single class rather than use this helper. I don't feel massively strongly here though

Reply:
I don't think that this matters. The aim of this function to just use a raw object rather than a class to avoid exposing private members in API event.